runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -414,6 +400,38 @@ public void evaluate(ParDo.BoundMulti<InputT, OutputT> transform, EvaluationCont
         }
       }
 
+      private void evaluateSingle(
```

Comment:
Why do we need this at all - won't evaluateMulti work well enough? If it's for performance reasons, please add a comment about that. And even then, perhaps there's a way to express the optimization by e.g. skipping the .filter() call in evaluateMulti in case there's only one output?